main.tf

```diff
@@ -78,7 +78,27 @@ resource "aws_codebuild_project" "default" {
     compute_type    = "${var.instance_size}"
     image           = "${var.image}"
     type            = "LINUX_CONTAINER"
-    privileged_mode = true
+    privileged_mode = "${var.privileged_mode}"
+
+    environment_variable {
+      "name"  = "AWS_REGION"
+      "value" = "${var.aws_region}"
```

Comment:
Use a ternary to populate the region from a data source if the region is empty.

https://www.terraform.io/docs/providers/aws/d/caller_identity.html